File: 2021/Day 1/lua/code.lua
function dofile(filename) return assert(loadfile(debug.getinfo(2, "S").source:sub(2):match("(.*/)") .. "/" .. filename))() end

local Array = dofile("./../../Common/lua/Array.lua")
local Utility = dofile("./../../Common/lua/Utility.lua")

local code = {}

function code:part1(input)
	local input = Utility.split(input, "\n", true)
	return input:filter(function(v, k, t)
		return k > 1 and v > t[k - 1]
	end).length
end

function code:part2(input)
	local input = Utility.split(input, "\n", true)
	return input:filter(function(v, k, t)
		return k > 1 and (v + (t[k + 1] or 0) + (t[k + 2] or 0)) > (t[k - 1] + v + (t[k + 1] or 0))
	end).length
end

return code

File: 2021/Day 1/Javascript/code.js
exports.part1 = (input) => input.split("\n").map(Number).filter((e, i, a) => i >= 1 && e > a[i-1]).length;


exports.part2 = (input) => input.split("\n").map(Number).filter((e, i, a) => i >= 1 && e + a[i + 1] + a[i + 2] > a[i-1] + e + a[i + 1]).length;